# map sampling factor-> The chance of getting proper room configuration depends on this value ------> in [m]
# 1. choice of value is very important
# 2. slightly large value will give better sampled map
# 3. Extra large value will cost more computation time
# 4. large value is needed for big map
# 5. small value is needed for small map

map_sampling_factor_check: 1.5

# room area factor-> Set the limitation of area of the room -------> in [m^2]
# 1. choice of value is very important
# 2. smaller value increase the chance to get every room

#morphological segmentation: 47.0 - 0.8
room_area_factor_upper_limit_morphological: 47.0
room_area_factor_lower_limit_morphological: 0.8

#distance segmentation: 163.0 - 0.35 (lower limit is perfect, upper limit is discussable)
room_area_factor_upper_limit_distance: 163.0
room_area_factor_lower_limit_distance: 0.35

#Voronoi segmentation: 120.0 - 1.53
room_area_factor_upper_limit_voronoi: 120.0
room_area_factor_lower_limit_voronoi: 1.53

#Semantic Segmentation: 23.0 - 1.0   (minimum and maximum room sizes in square meters)
room_area_factor_upper_limit_semantic: 23.0
room_area_factor_lower_limit_semantic: 1.0

#Voronoi random field segmentation: 1.53 - 1000000.0 
room_lower_limit_voronoi_random: 1.53
room_upper_limit_voronoi_random: 1000000.0

#parameters for the voronoi segmentation that specify the neighborhood for critical Point extraction 
#and the distance between critical Points. 
voronoi_neighborhood_index: 280 #larger value sets a larger neighborhood for searching critical points --> int
max_iterations: 150 #sets the maximal number of iterations to search for a neighborhood, also used for the vrf segmentation --> int
min_critical_point_distance_factor: 1.6 #minimal distance factor between two critical points before one of it gets eliminated --> double
max_area_for_merging: 12.5 #maximal area [mÂ²] of a room that should be merged with its surrounding rooms

#parameters for the voronoi random field segmentation that specify the size of the neighborhood generated on the Voronoi graph, the minimal
#size this neighborhood can have, how far base nodes for each node on the graph need to be apart and how many iterations the inference 
#max. should do
voronoi_random_field_epsilon_for_neighborhood: 7 #larger value sets larger neighborhood, concentrated in a node of the conditional random field --> int
min_neighborhood_size: 5 #min size of the above mentioned neighborhood --> int
min_voronoi_random_field_node_distance: 7.0 #min distance the base nodes for each crf node need to be apart --> double
max_voronoi_random_field_inference_iterations: 9000 #max number of iterations the inference algorithm should do --> int

#index = 280
#iterations = 150
#distance_factor = 1.6

# this variable selects the algorithm for room segmentation
# 1 = morphological segmentation
# 2 = distance segmentation
# 3 = Voronoi segmentation
# 4 = semantic segmentation
# 5 = voronoi random field segmentation
# int
room_segmentation_algorithm: 5

# displays the segmented map upon service call
# bool
display_segmented_map: false
